src/main/java/Ponder_Pika/Task/Deadline.java

```diff
@@ -0,0 +1,57 @@
+package Ponder_Pika.Task;
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
+
+
+/**
+ * This class represents a specific type of task that has a deadline.
+ * It extends the {@code Task} class and includes additional information about the deadline
+ * by which the task must be completed.
+ * The {@code Deadline} class overrides the following methods from the {@code Task} class:
+ * saveFullDetails() and toString()
+ */
+public class Deadline extends Task {
+    private final LocalDateTime deadline;
+
+    /**
+     * Constructs a Deadline task with the specified description and deadline.
+     *
+     * @param description the description of the task
+     * @param deadline the deadline by which the task should be completed
+     */
+    public Deadline(String description, LocalDateTime deadline) {
```

Comment:
I like the clear javadocs ! keep up the good work!

Reply:
Thank you for the review!